go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/prometheus/node_exporter
 
-go 1.22.0
+go 1.23.0
```

Comment:
What the reason of pinning explicit the .0 release instead of `go 1.23`

Reply:
Because the behavior of Go module version matching has changed since they started releasing new versions as `1.X.0` instead of how they used to release as `1.X` for the first version of a new major release.

Reply:
The go directory points to a minimum language version, not to an specific release version of go. A language version can optionally include a point release. `go 1.23` would fine as well. 

Example: https://github.com/prometheus-community/windows_exporter/blob/63efa92be7398883072dbb8fc15d11fc09f58621/go.mod#L3

Reply:
Yes, I'm aware. The update to the `golang.org/x/...` modules require >= 1.23.0.

The issue is just `go 1.23` confuses the toolchain, necessitating a toolchain directive. I don't want to include this here.